src/common/snippets/src/pass/propagate_precision.cpp

```diff
@@ -29,6 +29,14 @@ bool ov::snippets::pass::PropagatePrecision::run_on_model(const std::shared_ptr<
 
     bool was_updated = false;
     for (const auto& op : f->get_ordered_ops()) {
```

Comment:
@praasz  I think, I don't quite get you here. Are you talking about having an alternative to run_on_model(), but returning `void` or what? Could you elaborate on that, please? I don't really see how this is connected to the introduced changes. Thanks!

Reply:
The main idea is to find way to avoid code duplication. This PR adds code which looks very similar and could be extracted to some common function and then re-use it.

I mean to replace this code (and similar version(s)) added in this PR
```cpp
        if (const auto& multi_subgraph_op = std::dynamic_pointer_cast<op::util::MultiSubGraphOp>(node)) {
            for (const auto& sub_graph : multi_subgraph_op->get_functions()) {
                if (sub_graph) {
                    run_on_model(sub_graph);
                }
            }
        }
```
by some function which implement this pointer cast to MultiSubGraphOp and call run_on_model then just call it
```cpp
    my_run_model_helper(...);
```

I mention that there could be maybe two versions helper as in some places the return value of run_on_model is used to are set flag ```  rewritten |= run_on_model(sub_graph);```.

Reply:
Yeah, I've already discussed that and I am currently implementing this :)
The only thing, why to have two helper functions? I think, we may just ignore the return value in the places it's not used.

Reply:
Yes, it can be one it depends on implementation.

Reply:
Implemented the helper function. Should be good now.